components/ShareBike.tsx

```diff
@@ -0,0 +1,131 @@
+import { Bike } from '../lib/bike'
+import { Button } from './Button'
+import { useState, FormEvent } from 'react'
+import { FormError } from './Form'
+import { Api } from '../lib/api'
+
+export function ShareBike({bike, api}: {bike: Bike, api: Api}) {
+    const [successModal, setSuccessModal] = useState(false)
+    const [error, setError] = useState<string | undefined>(undefined)
+    const [shareinfo, setShareinfo] = useState({
+        email: "",
+        bikeId: bike.id,
+        role: "user",
+        duration: 1 // 86400 = 1 day, the current duration is just 1 second.
+    })
+
+    const ButtonStyling = {
+        "margin": "1rem"
+   }
+
+    const onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        try {
+            api.createBikeSharingInvitation(shareinfo)
+        } catch (e) {
+            setError(`${e}`)
+        }
+    }
+
+    return (
+        <div className='shareBike'>
+            <h3>Share bike</h3>
+
+            <form onSubmit={onSubmit}>
+                <div className="inputGroup">
+                    <label htmlFor="email">Email</label>
+                    <input 
+                        className = "input-mail"
+                        type = "email" 
+                        id = "email" 
+                        name = "email" 
+                        placeholder = "Enter email" 
+                        value = { shareinfo.email } 
+                        onChange = {
+                            e => setShareinfo(v => ({
+                                ...v, email: e.target.value
+                            }))
+                        }
+                        required 
+                    />
+
+                    <p>How many days would you like to share your bike?</p>
+                    <input 
+                        className = "input-number"
+                        type = "range" 
+                        id = "days" 
+                        name = "days" 
+                        step = "1"
+                        min  = "1" 
+                        max = "10" 
+                        list = "tickmarks"
+                        onChange = { 
+                            e => setShareinfo(v => ({
+                                ...v, duration: Number(e.target.value)
+                            }))
+                        }
+                        value = { shareinfo.duration }
+                    />
+
+                    <Button type="submit" style={ ButtonStyling }>
```

Comment:
the buttonStyleing is so small i think you can inline that value here
```suggestion
                    <Button type="submit" style={{margin: "1rem"}}>
```